state_solver/suspension.py

```diff
@@ -13,43 +13,53 @@
 
 class Suspension():
     """ Handles suspension vehicle forces """
-    def __init__(self, params:vehicle_params.BaseVehicle, logger:engine.Logger):
+    def __init__(self, params, logger:Logger):
         self.logger = logger
-        self.params:vehicle_params.BaseVehicle = params
-        self.__tires = engine.Tires(params)
+        self.params = params
+        self.__tires = Tires(params)
 
 
-    def get_loads(self, vehicle_velocity:np.array, yaw_rate:float, steered_angle:float, roll:float, pitch:float, heave:float):
-        forces, moments = np.array([0, 0, 0]), np.array([0, 0, 0])
+    def get_loads(self, vehicle_velocity:np.array, yaw_rate:float, steered_angle:float, roll:float, pitch:float, heave:float, slip_ratios:list):
+        veh_forces, veh_moments, wheel_speeds, wheel_torques = np.array([0, 0, 0]), np.array([0, 0, 0]), np.array([]), np.array([])
         
         # tire output forces as a function of inclination angle, slip angle, and normal force
         for tire_name, tire in self.__tires.items():
             
             ### ~~~ Normal Force Calculation ~~~ ###
             normal_force = self.__get_tire_normal_load(tire_name, tire, heave, pitch, roll)
-            
+
             ### ~~~ Slip Angle Calculation ~~~ ###
-            tire_velocity = vehicle_velocity + np.cross(np.array([0, 0, yaw_rate]), tire.position) # in IMF
+            tire_IMF_velocity = vehicle_velocity + np.cross(np.array([0, 0, yaw_rate]), tire.position) # in IMF
             steering_toe_slip = tire.steering_induced_slip(steered_angle)
-            slip_angle = np.arctan2(tire_velocity[1], tire_velocity[0]) +  steering_toe_slip# f(steered angle, body slip, yaw rate)
+            slip_angle = -np.arctan2(tire_IMF_velocity[1], tire_IMF_velocity[0]) +  steering_toe_slip# f(steered angle, body slip, yaw rate)
             
             ### ~~~ Inclination Angle Calculation ~~~ ###
             inclination_angle = self.__get_inclination_angle(tire_name, tire, steered_angle, roll, heave, pitch)
             
             ### ~~~ Tire Output Force Calculation ~~~ ###
-            f, m = self.__get_tire_output(tire_name, tire, normal_force, slip_angle, inclination_angle, steering_toe_slip)
-            forces = np.add(f, forces)  
-            moments = np.add(m, moments)
+            slip_ratio = tire.get_slip_ratio(slip_ratios)
+            tire_forces, tire_moments, tire_torque = self.__get_tire_output(tire_name, tire, normal_force, slip_angle, inclination_angle, steering_toe_slip, slip_ratio)
+            veh_forces = np.add(tire_forces, veh_forces)  
+            veh_moments = np.add(tire_moments, veh_moments)
+            wheel_torques = np.append(wheel_torques, [tire_torque])
+
+            ### ~~~ Wheel Speeds from Slip Ratios and Tire Velocities ~~~ ###
+            tire_pointing_unit_vector = np.array([np.cos(steering_toe_slip), np.sin(steering_toe_slip), 0])
```

Comment:
Not a fan of specifying a single unit vector per tire, would prefer to explicitly define the tire frame transform relative to the IMF. (Rot matrix from IMF to tire orientation, parameterized by slip angle).

Reply:
Pointing velocity also incidentally has an actual meaning (https://en.wikipedia.org/wiki/Poynting_vector), pronounced the same albeit spelled a little differently - naming a vector it can be confusing when talking about it in person (at least to me)

Reply:
You've been saying this one for a while, can you create a PR to fix it?

Reply:
yeah i'll flag it as an issue and start a PR after this is approved